.github/workflows/uitests.yml

```diff
@@ -0,0 +1,69 @@
+name: Build main target
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/**'
+      - NextcloudTalk.xcodeproj/**
+      - NextcloudTalk/**
+      - NotificationServiceExtension/**
+      - ShareExtension/**
+
+    push:
+      branches:
+        - main
+        - master
+        - stable*
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: macOS-latest
+    if: github.event.pull_request.draft == false
+
+    strategy:
+      matrix:
+        test-branches: ['stable27', 'master']
+
+    env:
+      WORKSPACE: NextcloudTalk.xcworkspace
+      DESTINATION: platform=iOS Simulator,name=iPhone 14,OS=16.2
+      SCHEME: NextcloudTalk
+      TEST_BRANCH: ${{ matrix.test-branches }}
+      CONTAINER_NAME: nextcloud_test
+      SERVER_PORT: 8080
+
+    steps:
+    - name: Checkout app
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+
```

Comment:
Aren't we cloning the branch here and then also cloning it in `ci-install-talk.sh`?

Reply:
No, this one (`actions/checkout`) checks out the talk-ios repository. The one in `ci-install-talk.sh` checks out the spreed (so the server part of talk) repository to install the app on the instance.